core/src/main/java/hudson/node_monitors/ResponseTimeMonitor.java

```diff
@@ -46,20 +46,23 @@
 public class ResponseTimeMonitor extends NodeMonitor {
     @Extension
     public static final AbstractNodeMonitorDescriptor<Data> DESCRIPTOR = new AbstractAsyncNodeMonitorDescriptor<Data>() {
+
         @Override
         protected Callable<Data,IOException> createCallable(Computer c) {
-            if (c.getChannel() == null) {
-                return null;
-            }
             return new Step1(get(c));
         }
 
         @Override
         protected Map<Computer, Data> monitor() throws InterruptedException {
-            Map<Computer, Data> base = super.monitor();
+            ResultMap<Data> base = (ResultMap<Data>) super.monitor();
             for (Entry<Computer, Data> e : base.entrySet()) {
                 Computer c = e.getKey();
                 Data d = e.getValue();
+                if (base.getSkipped().contains(c)) {
+                    assert d == null;
```

Comment:
> ... asserts are useful during development cycles but don't belong in production code.

That is the reason why Java they are turned on during unit testing and off in production.

My point here is to express expectation the computer in skipped set must have `null` as monitoring value. It will help when when eventual bug in the code decides to manifest its presence :P.

Reply:
I'm fine with that. I just wanted to check if you thought it would help with anything.